app/src/main/java/org/xbmc/kore/service/ConnectionObserversManagerService.java

```diff
@@ -67,35 +67,31 @@ public void onCreate() {
     @Override
     public int onStartCommand(Intent intent, int flags, int startId) {
         LogUtils.LOGD(TAG, "onStartCommand");
-        // Create the observers we are managing
-        createObservers();
-
-        // If no observers created, stop immediately
-        if (mConnectionObservers.isEmpty()) {
-            LogUtils.LOGD(TAG, "No observers, stopping observer service.");
-            stopSelf();
-            return START_NOT_STICKY;
-        }
 
         // Get the connection observer here, not on create to check if
         // there has been a change in hosts, and if so unregister the previous one
-        HostConnectionObserver connectionObserver = HostManager.getInstance(this).getHostConnectionObserver();
+        HostConnectionObserver connectionObserver = HostManager.getInstance(this)
+                .getHostConnectionObserver();
 
         // If we are already initialized and the same host, exit
         if (mHostConnectionObserver == connectionObserver) {
             LogUtils.LOGD(TAG, "Already initialized");
-            return START_STICKY;
+            stopForeground(true);
+            stopSelf();
```

Comment:
I fail to see the logic here. If the service is already running for the HostConnectionObserver we now stop it?

Reply:
You're right. My bad. Ignore this please.